scipy/optimize/_differentialevolution.py

```diff
@@ -544,6 +578,42 @@ def __init__(self, func, bounds, args=(),
         else:
             self.init_population_array(init)
 
+        # infrastructure for constraints
+        # dummy parameter vector for preparing constraints
+        x0 = self._scale_parameters(self.population[0])
+        self.constraints = []
+
+        if hasattr(constraints, '__len__'):
+            # sequence of constraints, this will also deal with default
+            # keyword parameter
+            for c in constraints:
+                try:
+                    feas = c.keep_feasible
+                    # need to switch off feasibility because it's not
+                    # guaranteed that any x0 is feasible
+                    c.keep_feasible = False
+                    self.constraints.append(PreparedConstraint(c, x0))
+                except Exception as e:
```

Comment:
the exception block is not covered by tests

Reply:
Exception would be raised if the PreparedConstraint construction doesn't work. I'd expect that to be covered by those tests.

Reply:
This is tested [here](https://github.com/scipy/scipy/blob/v1.2.1/scipy/optimize/tests/test_constraints.py#L77). I catch all `Exception`s here because I'm modifying the users constraint object, and want to definitely restore it. However, I'd only expect a `ValueError` to be raised.